src/pytest_inmanta_lsm/lsm_project.py

```diff
@@ -179,6 +188,51 @@ def lsm_services_update_attributes(
 
         return inmanta.protocol.common.Result(code=200, result={})
 
+    def lsm_services_update_attributes_v2(
+        self,
+        tid: uuid.UUID,
+        service_entity: str,
+        service_id: uuid.UUID,
+        current_version: int,
+        patch_id: str,
+        edit: typing.List[inmanta_lsm.model.PatchCallEdit],
+        comment: typing.Optional[str] = None,
+    ) -> inmanta.protocol.common.Result:
+        """
+        This is a mock for the lsm api, this method is called during allocation to update
+        the attributes of a V2 service.
+        """
+        # Making some basic checks
+        service = self.services[str(service_id)]
+        assert str(tid) == self.environment, f"{tid} != {self.environment}"
+        assert service.service_entity == service_entity, f"{service.service_entity} != {service_entity}"
+        assert service.version == current_version, f"{service.version} != {current_version}"
+
+        # The attributes parameter only represents the attributes that should be changed.
+        # * When no candidate attributes were set, the new candidate attributes will be equal to the active
+        #   attributes with the attribute updates applied.
+        # * When candidate attributes were set, the update will be applied to the existing candidate
+        #   attributes.
+        if service.candidate_attributes is None:
+            service.candidate_attributes = copy.deepcopy(service.active_attributes)
+
+        # Edit logic derived from:
+        # https://github.com/inmanta/inmanta-lsm/blob/39e9319381ce6cfc9fd22549e2b5a9cc7128ded2/src/inmanta_lsm/model.py#L2794
+
+        from inmanta_plugins.lsm.allocation_v2 import dict_path
```

Comment:
You shouldn't use this, but instead the dict_path library from inmanta_lsm directly

Reply:
https://github.com/inmanta/inmanta-lsm/blob/master/src/inmanta_lsm/dict_path.py

Reply:
Not this: https://github.com/inmanta/lsm/blob/master/plugins/allocation_v2/dict_path.py
(it is in the end the same code, but the second one is clearly just a backward compatibility fix)